src/clojure_lsp/handlers.clj

```diff
@@ -37,13 +37,13 @@
 
 (defmacro process-after-changes [& body]
   `(let [~'_time (System/nanoTime)]
-     (loop []
+     (loop [backoff# 1]
        (if (> (quot (- (System/nanoTime) ~'_time) 1000000) 60000) ; one minute timeout
          (log/warn "Timeout waiting for changes for body")
          (if (:processing-changes @db/db)
            (do
-             (Thread/sleep 100)
-             (recur))
+             (Thread/sleep backoff#)
+             (recur (min 200 (* 2 backoff#)))) ; 2^0, 2^1, ..., up to 200ms
```

Comment:
is there any chance we blow up the maximum value allowed for a long on backoff? I was just wordering, but probably only if we waited too many time which is acceptable I think

Reply:
oh not really, because of min, seems safe

Reply:
Yep, should be safe